inference-explainers/inference_onboarding_guide.md

```diff
@@ -0,0 +1,376 @@
+
+#### Step 1: Packaging
+
+1. **Build the GCP Confidential Space Docker image:** Follow the instructions in the aforementioned
+   [GCP Guide][3] to build a docker image for all B&A services. The inference artifacts, including
+   both the “tensorflow*v2_14_0” and “pytorch_v2_1_1” \_sidecar binaries* (ML runtimes and versions
+   that we currently support), will be packaged within the Bidding service image.
+    > **_Note:_** Other versions of Tensorflow and PyTorch will be supported in the future.
+
+#### Step 2: Deployment
+
+1. **Upload ML models:** Use a GCS bucket to host proprietary ML models. The Bidding server fetches
+   the uploaded models during runtime and forwards them to the inference sidecar for handling
+   requests. The bucket name is required by the Terraform configuration. Additionally, you must
+   upload a model configuration file that specifies the models being fetched to the cloud bucket and
+   include its path relative to the cloud bucket in the Terraform configuration. The model
+   configuration file has the following format:
+
+    ```
+    {
+        "model_metadata": [
+            {
+                "model_path": "model_1",
+                "checksum": "dd94b3b08ea19b4240aa5e5f68ff0447b40e37ebdd06cc76fa1a2cf61143a10c"
+            },
+            {
+                "model_path": "model_2",
+                "checksum": "<SHA-256 checksum value of hexadecimal string>"
+            }
+        ]
+    }
+    ```
+
+    Each model is specified by its path relative to the cloud bucket and may be accompanied by an
+    optional SHA-256 checksum of the model's content. The checksum can be computed by using the
+    following bash command:
+
+    ```
+    find <model_path> -type f -exec sha256sum {} \; | sort -k 2 | awk '{print $1}' | tr -d '\n' | sha256sum | awk '{print $1}'
+    ```
+
+    Store the metadata file in the same bucket as the models. You can update it after server
+    deployment to fetch additional models. Note that you must grant READ permission for your
+    proprietary endpoints to access your models and use them with your service.
+
+2. **Upload Code Modules:** The inference capabilities of your proprietary JavaScript bidding code
+   modules are exposed with the `runInference` and `getModelPaths` inference callbacks. To use
+   inference capabilities, update your existing code modules to invoke these functions. Refer to
+   [this section][7] of the B&A Inference Overview Explainer for more details about these two
+   functions, and to [this section][8] of the GCP Guide for more information about uploading code
+   modules.
+
+3. **Configure Terraform variables:** Configure the inference capabilities using inference Terraform
+   flags. The names of all inference-related Terraform flags have an `INFERENCE` prefix. Include
+   them as part of the `runtime_flags` in the “buyer” module.
+
+    3.1. Use `INFERENCE_SIDECAR_BINARY_PATH` to expose your ML runtime selection. To enable
```

Comment:
the sub points dont seem to be formatted properly ... you'd have to start the list again as 1. 2. etc. BTW, in many cases, people just write 1. 1. 1. for each of the bullets, since markdown does the numbering automatically.

Reply:
Thanks for pointing this out. Done.